test/util/thread.test.cpp

```diff
@@ -275,3 +275,57 @@ TEST(Thread, PauseResume) {
     thread.actor().invoke(&TestWorker::send, [&] { loop.stop(); });
     loop.run();
 }
+
+TEST(Thread, MultiplePauseResumeCalls) {
+    RunLoop loop;
+
+    Thread<TestWorker> thread("Test");
+
+    // Test if multiple pause calls work
+    thread.pause();
+    thread.pause();
+    thread.resume();
+    thread.resume();
+
+    thread.actor().invoke(&TestWorker::send, [&] { loop.stop(); });
+    loop.run();
+}
+
+TEST(Thread, TestImmediatePause) {
```

Comment:
This looks like a timing-dependent test, which almost inevitably seem to fail intermittently.